### PR TITLE
Add search by tag bar in feed page

### PR DESCRIPTION
**Pull request by emmanguyen28**

### CODE REVIEW

**Comment by nykekana** [comment]
Hi Emma. Did you modify the Message class as well?

**Comment by nykekana** [comment]
I'm so sorry! I closed it by accident!

**Review by nykekana — APPROVED** [review]
These are all good. Thanks Emma

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 1]
```diff
@@ -71,13 +71,14 @@ public void storeMessage(Message message) {
 				String text = (String) entity.getProperty("text");
 				long timestamp = (long) entity.getProperty("timestamp");
 				String imageUrl = (String) entity.getProperty("imageUrl");
+				String tag = (String) entity.getProperty("tag");
```
Comment (bluedragon2) [inline src/main/java/com/google/codeu/data/Datastore.java:90]: Could tag be null ?

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 2]
```diff
@@ -107,17 +120,18 @@ public void storeMessage(Message message) {
...
 	 */
-	public Set<String> getUsers(){
-		  Set<String> users = new HashSet<>();
-		  Query query = new Query("Message");
-		  PreparedQuery results = datastore.prepare(query);
-		  for(Entity entity : results.asIterable()) {
-		    users.add((String) entity.getProperty("user"));
-		  }
-		  return users;
+	public Set<String> getUsers() {
```
Comment (bluedragon2) [inline src/main/java/com/google/codeu/data/Datastore.java]: Are you retrieving all users on the system or all users that have posted a message ?
Imagine that you have a very popular app and there are 20 million messages per day = ~600 million messages a month. Making a call to getUsers will read all 600 messages just to get the names of the users.

If you want to get a list of all users in the app, then you can have a users table (or storage).
If you want the list of all users that have posted a message, then you can query on the user id (which will be indexed) which will be faster. You can use the .setDistinctOn("...") clause to get the set of distinct users.

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 1 above)
Comment (emmanguyen28) [reply src/main/java/com/google/codeu/data/Datastore.java:90]: Yes. When tag is null, the page simply loads all the messages

**Pull request merged**